lib/http.js

```diff
@@ -41,6 +41,8 @@ module.exports = function performFhServiceCall (opts, callback) {
       callback(null, null, body);
     }
   }
+  keycloak.getKeycloak().then(function(request) {
```

Comment:
This is pretty hacky. I'd prefer to add an option such as `injectHeadersFn` like so:

```js
const sync = mbaasSync({
  url: 'https://acme.example/com',
  timeout: 20000
  injectHeadersFn: () => {
    // This function returns a promise that should resolve with an object containing the headers, e.g
    return new Promise((resolve) => {
      // your token logic can go here...
      resolve({
        authorization: 'Bearer SOME_TOKEN'
      })
    })
  }
})
```

Then in your code above you can do:

```js
let extraHeaders = {}
if (opts.injectHeadersFn) {
  // Will need try/catch 
  extraHeaders = await opts.injectHeadersFn()
}
if (opts.headers) {
  opts.headers = Object.assign(extraHeaders, opts.headers)
} else {
  opts.headers = extraHeaders
}

request(opts, onServiceCallComplete)
```

Reply:
That's a much cleaner approach, thanks.